### ISSUE TITLE
[java] ConstructorCallsOverridableMethod - false-positive for final method call

### ISSUE THREAD

**Issue by Wolf2323** [pmd/pmd#5012]
**Affects PMD Version:** 7.0.0

**Rule:** ConstructorCallsOverridableMethod

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#constructorcallsoverridablemethod

**Description:**
With the update to PMD 7 this call to a final method in a constructor is now detected

**Code Sample demonstrating the issue:**

```java
	protected AbstractClause(@NotNull final Language defaultLanguage, @NotNull final String defaultTranslation) {
		add(Objects.requireNonNull(defaultLanguage), Objects.requireNonNull(defaultTranslation));
		this.defaultLanguage = defaultLanguage;
	}

	public @NotNull final T add(@NotNull final Language language, @NotNull final String translation) {
		if (translations.containsKey(Objects.requireNonNull(language))) {
			throw new UnsupportedOperationException("A translation for the supplied has already been added.");
		}

		translations.put(language, Objects.requireNonNull(translation));
		return getThis();
	}
```

**Expected outcome:**
The method is final, so there should be no violation

[INFO] PMD Failure: works.reload.recore.intl.AbstractClause:34 Rule:ConstructorCallsOverridableMethod Priority:1 This method may call an overridable method during object construction: `AbstractClause<T extends AbstractClause<T>>.getThis() (call stack: [AbstractClause<T extends AbstractClause<T>>.add(@NotNull Language, @NotNull String), AbstractClause<T extends AbstractClause<T>>.getThis()])`.

**Running PMD through:** *Maven*

**Comment by Wolf2323** [comment]
Oh i see, thats right
